src/interactive-window/interactiveWindow.node.ts

```diff
@@ -575,22 +578,15 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
                 }
             };
 
+            this.codeGeneratorFactory.getOrCreate(kernel).generateCode(cell, isDebug);
+
             // If debugging attach to the kernel but don't enable tracing just yet
             if (isDebug) {
                 await this.interactiveWindowDebugger.attach(kernel);
-
-                // Enable has to happen after the hidden code so that we don't hit breakpoints from previous cells
-                // Example:
-                // User has breakpoint on previous cell with name <ipython-2-hashystuff>
-                // We turn on tracing
-                // Hidden cell executes to set next cell to name <ipython-3-hashyotherstuff>
-                // Breakpoint fires in <ipython-2-hashystuff> because hidden cell inherits that value.
-                // So we have to enable tracing after we send the hidden cell.
-                kernelBeginDisposable = kernel.onPreExecute((c) => {
```

Comment:
All of this complication has been removed now, much simpler flow

Reply:
100%